### PR TITLE
Feature #60 edit user data with dot notation

### PR DESCRIPTION
**Pull request by theclayton**
Implements: https://gitlab.com/trazi-ventures/white-label/services/account/-/issues/60

### CODE REVIEW

File: include/routes/v1/functions/users.js [hunk 1]
```diff
@@ -101,17 +101,69 @@ async function editUser(opts) {
   var model = opts.req.body;
   var oldModel;
 
-  if (config.audit.edit.enable === true) {
-    oldModel = await getUser(opts);
+  let splitProp = [];
+
+  if (opts.req.params.prop) {
```
Comment (Phara0h) [inline include/routes/v1/functions/users.js:106]: Should check `indexOf('.') > -1`

File: include/routes/v1/functions/users.js [hunk 2]
```diff
@@ -101,17 +101,69 @@ async function editUser(opts) {
...
+    splitProp = opts.req.params.prop.split('.');
   }
 
+  if (config.audit.edit.enable === true || splitProp[0] === 'user_data') {
+    if (splitProp[0] === 'user_data' && opts.req.params.prop !== 'user_data') {
+      const params = opts.req.params;
+      const newOpts = Object.assign({}, opts);
+      // get user by id
+      newOpts.req.params = { id: opts.req.params.id, domain: opts.req.params.domain };
```
Comment (Phara0h) [inline include/routes/v1/functions/users.js:115]: Why are you doing all this extra allocation and processing? You could just pass in `{ id: opts.req.params.id, domain: opts.req.params.domain }` into getUser on line 117

Comment (theclayton) [reply include/routes/v1/functions/users.js]: Simplified in latest commit

File: include/routes/v1/functions/users.js (hunk 1 above)
Comment (theclayton) [reply include/routes/v1/functions/users.js]: added

**Pull request merged**